integration/data.go

```diff
@@ -97,37 +99,84 @@ func verifySeriesMapForRange(
 	start, end time.Time,
 	namespace string,
 	expected seriesList,
+	expectedDebugFilePath string,
+	actualDebugFilePath string,
 ) {
 	actual := make(seriesList, len(expected))
 	req := rpc.NewFetchRequest()
 	for i := range expected {
 		s := &expected[i]
 		req.NameSpace = namespace
-		req.ID = s.id
+		req.ID = s.ID
 		req.RangeStart = xtime.ToNormalizedTime(start, time.Second)
 		req.RangeEnd = xtime.ToNormalizedTime(end, time.Second)
 		req.ResultTimeType = rpc.TimeType_UNIX_SECONDS
 		fetched, err := ts.fetch(req)
 		require.NoError(t, err)
 		actual[i] = series{
-			id:   s.id,
-			data: fetched,
+			ID:   s.ID,
+			Data: fetched,
 		}
 	}
+
+	if len(expectedDebugFilePath) > 0 {
+		writeVerifyDebugOutput(t, expectedDebugFilePath, start, end, expected)
+	}
+	if len(actualDebugFilePath) > 0 {
+		writeVerifyDebugOutput(t, actualDebugFilePath, start, end, actual)
+	}
+
 	require.Equal(t, expected, actual)
 }
 
+func writeVerifyDebugOutput(t *testing.T, filePath string, start, end time.Time, series seriesList) {
+	w, err := os.OpenFile(filePath, os.O_APPEND|os.O_WRONLY, os.ModeAppend)
```

Comment:
This is an integration test component that is only enabled "when you want it".  It's for viewing the difference between expected and actual datapoints that arrived at the destination node.  It needs to live on after the test run is finished so you can do a diff between the two files.